integration/spark/src/main/java/org/apache/spark/sql/secondaryindex/query/SecondaryIndexQueryResultProcessor.java

```diff
@@ -294,25 +289,31 @@ private void processResult(List<CarbonIterator<RowBatch>> detailQueryResultItera
         preparedRow[i] = wrapper.getDictionaryKeyByIndex(dictionaryIndex++);
       } else {
         if (isComplexColumn) {
```

Comment:
Please add a new check before this, for add column scenario

Reply:
ok